apps/zbugs/shared/queries.ts

```diff
@@ -41,19 +42,51 @@ const issueRowSort = z.object({
 
 type IssueRowSort = z.infer<typeof issueRowSort>;
 
+const defineAuthQuery = <
+  Name extends string,
+  Input extends ReadonlyJSONValue | undefined,
+  Output extends ReadonlyJSONValue | undefined,
+  TSchema extends Schema,
+  TTable extends keyof TSchema['tables'] & string,
+  TReturn,
+>(
+  name: Name,
+  validator: StandardSchemaV1<Input, Output>,
+  queryFn: (input: {
+    args: Output;
+    ctx: AuthData | undefined;
+  }) => Query<TSchema, TTable, TReturn>,
+) => defineQuery(name, {validator}, queryFn);
+
 export const queries = {
-  allLabels: syncedQuery('allLabels', z.tuple([]), () => builder.label),
+  allLabels: defineQuery(
+    'allLabels',
+    {
+      validator: z.undefined(),
+    },
+    () => builder.label,
+  ),
 
-  allUsers: syncedQuery('allUsers', z.tuple([]), () => builder.user),
+  allUsers: defineQuery(
+    'allUsers',
+    {
+      validator: z.undefined(),
+    },
+    () => builder.user,
+  ),
 
-  user: syncedQuery('user', idValidator, userID =>
-    builder.user.where('id', userID).one(),
+  user: defineQuery(
+    'user',
+    {
+      validator: idValidator,
+    },
+    ({args: userID}) => builder.user.where('id', userID).one(),
   ),
 
-  issuePreload: syncedQueryWithContext(
+  issuePreload: defineAuthQuery(
     'issuePreload',
     idValidator,
-    (auth: AuthData | undefined, userID) =>
+    ({args: userID, ctx: auth}) =>
```

Comment:
here args get inferred from the validator and ctx is hard coded in defineAuthQuery.